tests/test_collection.py

```diff
@@ -535,6 +535,7 @@ def test_shape_driver_rounds_float_property_to_requested_digits_number(self):
             assert 12.2 == rf1.properties["property1"]
             assert 12.9 == rf2.properties["property1"]
 
+    @pytest.mark.xfail(reason="Corner case not supported by new feature builder implementation")
```

Comment:
Currently, I'm xfail-ing 4 tests that assert some weird behavior. For example, I'm not sure that Fiona should be ensuring that when you write `"3.14"` (a string) to an int field, you get a `3` (int) in your shapefile. This would require wrapping the setter's input value like `int(float(value))` and that's not negligible:

```
python3 -m timeit "int(3.14)"
10000000 loops, best of 5: 26.8 nsec per loop
python3 -m timeit "int(float('3.14'))"
5000000 loops, best of 5: 53.5 nsec per loop
```

Reply:
Oh, now I understand the difference between the new and old code. Previously, we dispatched to `OGR_F_SetFieldXXX()` based on the type of the value, not on the type of the field. I may try that.